packages/cosmic-swingset/lib/ag-solo/vats/lib-wallet.js

```diff
@@ -75,28 +76,42 @@ export async function makeWallet(
     // === AWAITING TURN ===
     // =====================
 
-    assert(
-      zoeKind === 'indexed',
-      details`Only indexed Zoe is implemented, not ${zoeKind}`,
-    );
-
     // We now have everything we need to provide Zoe, so do the actual withdrawal.
-    // purses are an array ordered by issuer payoutRules.
-    const payment = await Promise.all(
-      offerRules.payoutRules.map(({ kind, amount }, i) => {
-        const purse = purses[i];
-        if (kind === 'offerAtMost' && purse) {
-          return E(purse).withdraw(amount);
-        }
-        return undefined;
-      }),
-    );
+    let payment;
+    // console.log('FIGME: create payment from', offerRules);
+    if (zoeKind === 'roles') {
+      // Payments are made for the roles in offerRules.offer.
+      payment = {};
+      await Promise.all(
+        Object.entries(offerRules.offer || {}).map(([role, amount]) => {
+          const purse = purses[role];
+          if (purse) {
+            payment[role] = E(purse).withdraw(amount);
+          }
+          return payment[role];
+        }),
+      );
+    } else if (zoeKind === 'indexed') {
+      // purses/payment are an array indexed by issuer payoutRules.
+      payment = await Promise.all(
+        offerRules.payoutRules.map(({ kind, amount }, i) => {
+          const purse = purses[i];
+          if (kind === 'offerAtMost' && purse) {
+            return E(purse).withdraw(amount);
+          }
+          return undefined;
+        }),
+      );
+    } else {
+      assert(false, details`Unsupported zoeKind ${zoeKind}`);
```

Comment:
When is `assert(false, ... )` better than a direct `throw()`?

Reply:
When we need to hide from the caller what the `details` arguments are.  We don't in this case, so I'll change it to a throw.